src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -114,6 +115,25 @@ impl<'a, L: Language> CodeGenerator<'a, L> {
         ))
     }
 
+    /// Represents a [tera] filter that converts given tera string value to [snake case].
+    /// Returns error if the given value is not a string.
+    ///
+    /// For more information: <https://docs.rs/tera/1.19.0/tera/struct.Tera.html#method.register_filter>
+    ///
+    /// [tera]: <https://docs.rs/tera/latest/tera/>
+    /// [upper camel case]: <https://docs.rs/convert_case/latest/convert_case/enum.Case.html#variant.Snake>
+    pub fn snake(
+        value: &tera::Value,
+        _map: &HashMap<String, tera::Value>,
+    ) -> Result<tera::Value, tera::Error> {
+        Ok(tera::Value::String(
+            value
+                .as_str()
+                .ok_or(tera::Error::msg("Required string for this filter"))?
```

Comment:
Include the filter name in the error message:
```suggestion
                .ok_or(tera::Error::msg("the `snake` filter only accepts strings"))?
```

Alternatively, make `value` a `&str` so the caller (and not this string-specific code) has to deal with unexpected types.

Reply:
This [`as_str`](https://docs.rs/tera/1.19.1/tera/enum.Value.html#method.as_str) here is a method on `tera::Value`. It converts a tera value as string similar to `Element::as_str()`, which is why I have used `as_str` on these filters (`snake` or `upper_camel`).

Reply:
Right, but using that method forces this function to have to handle/propagate an error if `as_str` returns `None`. I was suggesting taking a `&str` as an argument so whoever calls this has to handle the not-a-string case.

Reply:
Since this method or filter is used inside the tera templates it is not possible to change the method signature here. tera filters follow a specific signature: https://docs.rs/tera/1.19.0/tera/trait.Filter.html

Reply:
Ah, got it! That makes sense, thanks for explaining.